spec/factories/author.rb

```diff
@@ -1,12 +1,33 @@
 FactoryBot.define do
   sequence(:random_id) do |n|
-    @random_ids ||= (10_000..1_000_000).to_a.shuffle
-    @random_ids[n]
+    @random_ids ||= {}
+    @random_ids[n] ||= begin
+      x = nil
+      while x.nil?
+        i = SecureRandom.random_number(990_000) + 10_000
+        x = i unless @random_ids.values.include?(i)
```

Comment:
This is a more reasonable approach, why not just go ahead and just put the value in as the hash as the key.  That's all we care about, not `n`.